pkg/ccl/backupccl/backup_job.go

```diff
@@ -402,14 +398,71 @@ func (b *backupResumer) Resume(ctx context.Context, execCtx interface{}) error {
 	details := b.job.Details().(jobspb.BackupDetails)
 	p := execCtx.(sql.JobExecContext)
 
-	var backupManifest *BackupManifest
+	// Resolve the backup destination. We can skip this step if we have already
+	// resolved and persisted the destination either during planning or a previous
+	// resumption of this job.
+	defaultURI := details.URI
+	var backupDest backupDestination
+	if details.URI == "" {
+		var err error
+		backupDest, err = resolveDest(ctx, p.User(), details.Destination, details.EndTime,
+			details.IncrementalFrom, p.ExecCfg())
+		if err != nil {
+			return err
+		}
+		defaultURI = backupDest.defaultURI
+	}
+
+	// The backup job needs to lay claim to the bucket it is writing to, to
+	// prevent concurrent backups from writing to the same location.
+	//
+	// If we have already locked the location, either on a previous resume of the
+	// job or during planning because `clusterversion.BackupResolutionInJob` isn't
+	// active, we do not want to lock it again.
+	foundLockFile, err := checkForBackupLockFile(ctx, p.ExecCfg(), defaultURI, b.job.ID(), p.User())
+	if err != nil {
+		return err
+	}
+
+	// TODO(adityamaru): We can delete the `details.URI == ""` check in 22.2. This
+	// is present in 22.1 to guard against the case where we have already written
+	// a BACKUP-LOCK file during planning (mixed-version cluster) and do not want
+	// to re-check and re-write the lock file. In that case `details.URI` will
+	// non-empty.
+	if details.URI == "" && !foundLockFile {
```

Comment:
@stevendanna calling this out as something I'd like a sanity check on. This relates to the other comment of no longer needing the bool.

Reply:
The bool I was referring to was the previously present `HasLockedLocation`.

> if details.URI is empty, how can we have found a lock file?

I think this is possible if a backup job has resolved the destination on the first resumption, written a lock file, but **not** updated the job details with the resolved destination before it gets re-resumed. In which case on the second resume, `details.URI == ""` but we would find a lock file?

Reply:
Discussed offline: we need this because of having moved a bunch of code that was formerly in backup planning to backup job, and we need to support interruptions/restarts in both versions of the code.